### PR TITLE
Test the `task name` subcommand

### PR DESCRIPTION
**Pull request by djmitche**
My proposal for how we should go about testing subcommands.  H/t to Ayub for the example of asserting on subcommand output :)

### CODE REVIEW

File: cmds/task/fetchers_test.go [hunk 1]
```diff
@@ -2,12 +2,59 @@ package task
...
+}
+
+func (suite *FakeServerSuite) TestNameCommand() {
+	// set up to run a command and capture output
+	buf := &bytes.Buffer{}
+	cmd := &cobra.Command{}
+	cmd.SetOutput(buf)
+
+	// run the command
+	args := []string{"ANnmjMocTymeTID0tlNJAw"}
```
Comment (walac) [inline cmds/task/fetchers_test.go]: nit: can we make this a `const taskId = "ANnmjMocTymeTID0tlNJAw"`?

**Comment by t0xicCode** [comment]
It would be good to check why TC failed on this though.

**Comment by djmitche** [comment]
@lteigrob have you seen this from gometalinter before?
https://tools.taskcluster.net/task-group-inspector/#/cWw_GnD5SSKnQAun-HNFOg/cWw_GnD5SSKnQAun-HNFOg?_k=44gbb9
```
WARNING: deadline exceeded by linter gotype on cmds/config (try increasing --deadline)
WARNING: deadline exceeded by linter gotype on . (try increasing --deadline)
WARNING: deadline exceeded by linter gosimple on cmds/task (try increasing --deadline)
WARNING: deadline exceeded by linter gotype on client (try increasing --deadline)
WARNING: deadline exceeded by linter gotype on cmds/from-now (try increasing --deadline)
WARNING: deadline exceeded by linter gotype on cmds/root (try increasing --deadline)
WARNING: deadline exceeded by linter gotype on cmds/group (try increasing --deadline)
WARNING: deadline exceeded by linter gotype on cmds/signin (try increasing --deadline)
WARNING: deadline exceeded by linter gotype on cmds/task (try increasing --deadline)
WARNING: deadline exceeded by linter gotype on cmds/version (try increasing --deadline)
WARNING: deadline exceeded by linter deadcode on . (try increasing --deadline)
WARNING: deadline exceeded by linter golint on client (try increasing --deadline)
WARNING: deadline exceeded by linter gotype on config (try increasing --deadline)
WARNING: deadline exceeded by linter gotype on cmds/slugid (try increasing --deadline)
WARNING: deadline exceeded by linter golint on cmds/from-now (try increasing --deadline)
WARNING: deadline exceeded by linter golint on cmds/config (try increasing --deadline)
WARNING: deadline exceeded by linter golint on cmds/slugid (try increasing --deadline)
WARNING: deadline exceeded by linter golint on cmds/root (try increasing --deadline)
WARNING: deadline exceeded by linter golint on cmds/group (try increasing --deadline)
WARNING: deadline exceeded by linter golint on cmds/signin (try increasing --deadline)
WARNING: deadline exceeded by linter golint on config (try increasing --deadline)
WARNING: deadline exceeded by linter deadcode on cmds/root (try increasing --deadline)
WARNING: deadline exceeded by linter golint on cmds/version (try increasing --deadline)
WARNING: deadline exceeded by linter golint on cmds/task (try increasing --deadline)
WARNING: deadline exceeded by linter deadcode on client (try increasing --deadline)
WARNING: deadline exceeded by linter deadcode on cmds/from-now (try increasing --deadline)
WARNING: deadline exceeded by linter deadcode on cmds/task (try increasing --deadline)
WARNING: deadline exceeded by linter deadcode on cmds/signin (try increasing --deadline)
WARNING: deadline exceeded by linter deadcode on cmds/group (try increasing --deadline)
WARNING: deadline exceeded by linter deadcode on config (try increasing --deadline)
WARNING: deadline exceeded by linter deadcode on cmds/slugid (try increasing --deadline)
WARNING: deadline exceeded by linter deadcode on cmds/config (try increasing --deadline)
WARNING: deadline exceeded by linter deadcode on cmds/version (try increasing --deadline)
```

I'll re-run the task by closing/reopening..

**Comment by djmitche** [comment]
Let's see if that helps...

**Comment by djmitche** [comment]
actual lint that time -- trying again

**Comment by djmitche** [comment]
@lteigrob what do you think about the deadline change?  I'm not sure why that was required (but it might have to do with not skipping `vendor`, which probably takes a long time to lint?)

**Comment by djmitche** [comment]
We just chatted about the deadline change -- sounds like it's a fine thing to add, and we've probably been on the edge of needing it for a while.

**Pull request merged**